File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03829/p03829_num3_parsed.cpp
#include <iostream>
#include <vector>
 
int main() {
    int N, A, B;
    std::cin >> N >> A >> B;
    
    std::vector<int> X(N);
    for (int i = 0; i < N; i++) {
        std::cin >> X[i];
    }
    
    long long total_fatigue = 0;
    for (int i = 0; i < N - 1; i++) {
        int distance = X[i + 1] - X[i];
        total_fatigue += std::min(distance * A, B);
    }
    
    std::cout << total_fatigue << std::endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03829/p03829_num3_parsed.py
import sys

def main():
    # Read input
    n, a, b = map(int, input().split())
    #  read X values as list
    X = list(map(int, input().split()))

    # Initialize fatigue level
    fatigue = 0

    # Iterate over towns
    for i in range(n-1):
        # Calculate distance between current town and next town
        distance = X[i+1] - X[i]
        # Calculate fatigue increase if walk
        walk_fatigue = distance * a
        
        # Calculate fatigue increase if teleport
        teleport_fatigue = b
        
        # Take minimum of walk and teleport fatigue
        fatigue += min(walk_fatigue, teleport_fatigue)
    
    # Print the minimum possible total increase of your fatigue level
    print(fatigue)


if __name__ == "__main__":
    main()